src/workflows-app/components/ImportGithub.js

```diff
@@ -35,61 +46,163 @@ const ImportGithub = ({ setLoading, signal, onDismiss, workspace }) => {
   const updateWorkflowName = (url) => {
     setMethodName(url.substring(url.lastIndexOf('/') + 1).replace('.wdl', ''));
   };
-  return div({ style: { marginLeft: '4rem', width: '50%' } }, [
-    div({ style: { fontSize: 30, display: 'flex', alignItems: 'center' } }, [
-      h(FormLabel, { htmlFor: 'methodurl', required: true }, ['Workflow Link']),
-      h(TooltipCell, { tooltip: 'Link must start with https://github.com or https://raw.githubusercontent.com' }, [
-        icon('error-standard', { size: 20, style: { top: '50px', marginLeft: '1rem', color: colors.accent(), cursor: 'help' } }),
-      ]),
-    ]),
-    h(ValidatedInput, {
-      inputProps: {
-        id: 'methodurl',
-        placeholder: 'Paste Github link',
-        value: methodUrl,
-        onChange: (u) => {
-          updateWorkflowName(u);
-          setMethodVersionName(getMethodVersionName(u));
-          setMethodUrl(u);
-          setMethodUrlModified(true);
-        },
-      },
-      error: Utils.summarizeErrors(methodUrlModified && errors?.methodUrl),
-    }),
-    h(FormLabel, { htmlFor: 'workflowName', required: true }, ['Workflow Name']),
-    h(ValidatedInput, {
-      inputProps: {
-        id: 'workflowName',
-        placeholder: 'Workflow Name',
-        value: methodName,
-        onChange: (n) => {
-          setMethodName(n);
-          setMethodNameModified(true);
-        },
-      },
-      error: Utils.summarizeErrors(methodNameModified && errors?.methodName),
-    }),
-    div({}, [
-      h(
-        ButtonPrimary,
-        {
-          style: { marginTop: '2rem' },
-          'aria-label': 'Add to Workspace button',
-          disabled: errors,
-          onClick: () => {
-            const method = {
-              method_name: methodName,
-              method_version: methodVersionName,
-              method_url: methodUrl,
-              method_source: 'GitHub',
-            };
-            withBusyState(setLoading, submitMethod(signal, onDismiss, method, workspace));
+
+  const onSuccess = (methodObject) => {
+    if (!isFeaturePreviewEnabled(ENABLE_WORKFLOWS_SUBMISSION_UX_REVAMP)) {
+      Nav.goToPath('workspace-workflows-app-submission-config', {
+        name: workspace.workspace.name,
+        namespace,
+        methodId: methodObject.method_id,
+      });
+    } else {
+      setMethodId(methodObject.method_id);
+      setImportLoading(false);
+      setSuccessfulImport(true);
+    }
+  };
+  const onError = async (error) => {
+    if (!isFeaturePreviewEnabled(ENABLE_WORKFLOWS_SUBMISSION_UX_REVAMP)) {
+      notify('error', 'Error creating new method', { detail: error instanceof Response ? await error.text() : error });
+      onDismiss();
+    } else {
+      setImportLoading(false);
+      setSuccessfulImport(false);
+      setErrorMessage(JSON.stringify(error instanceof Response ? await error.text() : error, null, 2));
+    }
+  };
+  return !isFeaturePreviewEnabled(ENABLE_WORKFLOWS_SUBMISSION_UX_REVAMP)
```

Comment:
really a nit pick, but if were me, I'd give these two divs their own function to make the return statement easier to read:

`return !isFeaturePreviewEnabled(ENABLE...) ? renderGithubImportClassic() : renderGithubImportRevamp();`

Reply:
Normally I'd agree, but since this is behind a feature flag and it's going away very soon, I think it should be alright for now.